enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions
+
+#### Can we use sqlite?
+
+It would be advantageous to use sqlite for the database especially on
+stand-alone clusters, so that it is not necessary to deploy and manage
+an entire RDBMS, nor ship and support its container image. The [gorm library
+supports sqlite](https://gorm.io/docs/connecting_to_the_database.html#SQLite),
+but it is not clear if assisted service is compatible. In particular, the [use
+of FOR
+UPDATE](https://github.com/openshift/assisted-service/blob/e70af7dcf59763ee6c697fb409887f00ab5540f5/pkg/transaction/transaction.go#L8)
+might be problematic.
+
+For Hub clusters doing multi-cluster creation and management, there is an
+expectation that persistent storage availability and scale concerns will be a
+better fit for running a full RDBMS.
+
+A suggestion has been made that in situations where there is a single process
+running the assisted service, locking can happen in-memory instead of in the
+database. Further analysis is required.
+
+#### baremetal-operator watching multiple namespaces?
+
+When utilizing baremetal-operator on a Hub cluster to boot the discovery ISO on
+hosts, should we be creating those BareMetalHost resources in a separate
+namespace from those that are associated with the Hub cluster itself?
+
+What work is involved in having BMO watch additional namespaces?
+
+Upstream, the metal3 project is already running baremetal-operator watching
+multiple namespaces, so there should not be software changes required.
+
+#### Centralized Machine Management
+
+OpenShift's multi-cluster management is moving toward a centralized Machine
+management pattern, as an addition to the current approach where Machines only
+exist in the same cluster as their associated Node. This proposal should be
+compatible with centralized Machine management, but it would be useful to play
+through those workflows in detail to be certain.
+
+#### Garbage Collection
+
+Agent resources are not useful after provisioning except possibly as a
+historical record. A plan should be in place for garbage collecting them.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
```

Comment:
I know we often don't spend a lot of time on this section, but I think it could be useful here

e.g. our plans around feature gates, API versions, etc.